### PR TITLE
Notes from an end to end run through on Windows

### PR DESCRIPTION
**Pull request by bobturneruk**
- [x] Setup
- [x] Ep1
- [x] Ep2
- [x] Ep3
- [x] Ep4
- [x] Ep5

### CODE REVIEW

**Comment by bobturneruk** [comment]
I've been pretty harsh with ep5. But this is now good to go. I hope.

File: _episodes/02-working-with-environments.md [hunk 1]
```diff
@@ -359,23 +354,23 @@ $ conda install scikit-learn=1.0.2
...
->
-> If you are working through these lessons using a Binder instance, then the default location of
-> the Conda environments is slightly different.
+> ## Finding Conda
 >
+> If you're not sure where your Conda environments are stored, activate an environment, launch Python, then run
+> the following:
+> ```
+> import sys                   
+> sys.executable
```
Comment (ns-rse) [inline _episodes/02-working-with-environments.md:369]: Would it be worth showing sample output here as well?

```
>>> import sys
>>> sys.executable
'/home/neil/miniconda3/bin/python'
```

File: _episodes/03-using-packages-and-channels.md [hunk 2]
```diff
@@ -336,6 +295,11 @@ $ which pip # sometimes installed as pip3
 ~~~
 {: .language-bash}
 
+> ## Windows users...
+>
+> You can type `where.exe` in **PowerShell** and it does the same thing as `which` in **bash**.
```
Comment (ns-rse) [inline _episodes/03-using-packages-and-channels.md:300]: Thanks, I lack experience with PowerShell.

File: _episodes/04-sharing-environments.md [hunk 3]
```diff
@@ -134,6 +133,17 @@ environment streamed to the terminal. Recall that we only listed five packages w
...
+> ## What's in the exported environment.yml file
+>
+> The exported version of the file looks a bit different to the one we wrote. In addition to version
+> numbers, on some lines we've got another code in there e.g. `vs2015_runtime=14.34.31931=h4c5c07a_10`. The `h4c5c07a_10`
+> is the [build variant hash](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#differentiating-packages-built-with-different-variants). This appears when the package is different
+> for different operating systems. The implication is that an environment file that contains a build variant
+> hash for one or more of the packages cannot be used on a different operating system to the one it
+> was created on.
+>
+{: .callout}
```
Comment (ns-rse) [inline _episodes/04-sharing-environments.md:145]: This is a really good point and I think we need to make it clearer in the later section where its stated...

> In short: to make sure others can reproduce your environment independent of the operating system they use, make sure to add the `--from-history` argument to the `conda env export` command.

...as this output may on some occasions include the hash and we should perhaps then advise to remove those if its going to be used across OS's.

File: _episodes/05-managing-cuda-dependencies.md [hunk 4]
```diff
@@ -118,163 +112,84 @@ cudnn                          7.6.5       cuda9.2_0  pkgs/main
...
-> 
-> Also take note of the channel priorities: the official `pytorch` channel must be given priority 
-> over `conda-forge` in order to insure that the official PyTorch binaries (the ones that include 
-> NCCL and cuDNN) will be installed (otherwise you will get some unofficial version of PyTorch 
-> available on `conda-forge`).
-{: .callout}
+We might want to check this has worked:
+
+```
+python -c "import torch; print(torch.cuda.is_available())"
```
Comment (ns-rse) [inline _episodes/05-managing-cuda-dependencies.md:151]: Do you think we need to explain what this is doing?

File: _episodes/05-managing-cuda-dependencies.md [hunk 5]
```diff
@@ -118,163 +112,84 @@ cudnn                          7.6.5       cuda9.2_0  pkgs/main
...
+python3 -m pip install "tensorflow<2.11"
+```
+
+Note the lack of support for recent versions on Windows. The [Windows Subsystem for Linux](https://learn.microsoft.com/en-gb/windows/wsl/install)
+is a good choice for installing later versions.
+
+We might want to check this has worked:
+
+```
+python -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"
```
Comment (ns-rse) [inline _episodes/05-managing-cuda-dependencies.md:185]: As above, do you think we need to explain this in greater detail?

File: setup.md [hunk 6]
```diff
@@ -113,11 +113,13 @@ rm Miniconda3-latest-Linux-x86_64.sh
 
 ## Verifying your Conda installation
 
+> On **Windows** you'll need to go to the "start" menu (or whatever it's called now) and open up the **Anaconda Powershell Prompt** before following the next instructions.
```
Comment (ns-rse) [inline setup.md:116]: > (or whatever it's called now)

:laughing:

**Review by ns-rse — APPROVED** [review]
Excellent work overhauling section 5, not an area I've much/any familiarity with.

I think we need to rework episodes 3 and 4 with regards to installing `pip` which if I recall one of your comments may have said comes bundled with Python now, but lets do that in the next iteration.

File: _episodes/02-working-with-environments.md (hunk 1 above)
Comment (bobturneruk) [reply _episodes/02-working-with-environments.md:369]: Yes - good idea.

File: _episodes/04-sharing-environments.md (hunk 3 above)
Comment (bobturneruk) [reply _episodes/04-sharing-environments.md:145]: Good point. I hope I've now got this covered.

File: _episodes/05-managing-cuda-dependencies.md (hunk 4 above)
Comment (bobturneruk) [reply _episodes/05-managing-cuda-dependencies.md:151]: Good plan.

File: _episodes/05-managing-cuda-dependencies.md (hunk 5 above)
Comment (bobturneruk) [reply _episodes/05-managing-cuda-dependencies.md:185]: Yup.

**Comment by bobturneruk** [comment]
Thanks for the feedback @ns-rse. I'll get this merged and see if we think the built version looks good to go.

**Pull request merged**